Build failure: sane-frontends

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix-build -A sane-frontends`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
scanadf.c:1604:20: error: passing argument 2 of 'signal' from incompatible pointer type [-Wincompatible-pointer-types]
 1604 |   signal (SIGPIPE, sighandler);
      |                    ^~~~~~~~~~
      |                    |
      |                    int (*)(int)
/nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/signal.h:88:57: note: expected '__sighandler_t' {aka 'void (*)(int)'} but argument is of type 'int (*)(int)'
   88 | extern __sighandler_t signal (int __sig, __sighandler_t __handler)
      |                                          ~~~~~~~~~~~~~~~^~~~~~~~~
scanadf.c:150:1: note: 'sighandler' declared here
  150 | sighandler (int signum)
      | ^~~~~~~~~~
/nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/signal.h:72:16: note: '__sighandler_t' declared here
   72 | typedef void (*__sighandler_t) (int);
      |                ^~~~~~~~~~~~~~
scanadf.c:1605:20: error: passing argument 2 of 'signal' from incompatible pointer type [-Wincompatible-pointer-types]
 1605 |   signal (SIGTERM, sighandler);
      |                    ^~~~~~~~~~
      |                    |
      |                    int (*)(int)
/nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/signal.h:88:57: note: expected '__sighandler_t' {aka 'void (*)(int)'} but argument is of type 'int (*)(int)'
   88 | extern __sighandler_t signal (int __sig, __sighandler_t __handler)
      |                                          ~~~~~~~~~~~~~~~^~~~~~~~~
scanadf.c:150:1: note: 'sighandler' declared here
  150 | sighandler (int signum)
      | ^~~~~~~~~~
/nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/signal.h:72:16: note: '__sighandler_t' declared here
   72 | typedef void (*__sighandler_t) (int);
      |                ^~~~~~~~~~~~~~
make[1]: *** [Makefile:68: scanadf.o] Error 1
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2, NixOS, 26.05 (Yarara), 26.05.20251228.c0b0e0f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - nixpkgs: `/nix/store/xzl5dhddgjxp2p18h5qpkhvlkp8128j1-source`


### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/graphics/sane/frontends.nix
+++ b/pkgs/applications/graphics/sane/frontends.nix
@@ -3,6 +3,7 @@
   stdenv,
   fetchurl,
   fetchpatch,
+  autoconf,
   sane-backends,
   libX11,
   gtk2,
@@ -67,9 +68,17 @@ stdenv.mkDerivation rec {
     gtk2
   ]
   ++ lib.optional (libusb-compat-0_1 != null) libusb-compat-0_1;
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [
+    pkg-config
+    autoconf
+  ];
 
   enableParallelBuilding = true;
+
+  # https://bugzilla.redhat.com/show_bug.cgi?id=2341321
+  preConfigure = ''
+    autoconf
+  '';
 
   meta = {
     description = "Scanner Access Now Easy";

--- a/pkgs/by-name/xs/xsane/package.nix
+++ b/pkgs/by-name/xs/xsane/package.nix
@@ -35,13 +35,17 @@ stdenv.mkDerivation rec {
         { name, hash }:
         fetchpatch {
           inherit name hash;
-          url = "https://src.fedoraproject.org/rpms/xsane/raw/846ace0a29063335c708b01e9696eda062d7459c/f/${name}";
+          url = "https://src.fedoraproject.org/rpms/xsane/raw/67c7f6591398ab46f71ba4f0eb33cb0539e56a4b/f/${name}";
         };
     in
     map fetchFedoraPatch [
       {
         name = "0001-Follow-new-convention-for-registering-gimp-plugin.patch";
         hash = "sha256-yOY7URyc8HEHHynvdcZAV1Pri31N/rJ0ddPavOF5zLw=";
+      }
+      {
+        name = "0001-replace-gtk_timeout-with-g_timeout.patch";
+        hash = "sha256-OMeukfbpagK05HI2R/2Gxrvk2nfrDK8ZmLmw0a6o10k=";
       }
       {
         name = "xsane-0.995-close-fds.patch";
@@ -106,6 +110,10 @@ stdenv.mkDerivation rec {
       {
         name = "xsane-configure-c99.patch";
         hash = "sha256-ukaNGgzCGiQwbOzSguAqBIKOUzXofSC3lct812U/8gY=";
+      }
+      {
+        name = "xsane-gcc15-c23.patch";
+        hash = "sha256-a1ZkVufpbwqL/bCFE2w44PoR42chl8aoRqeOP6L74Uw=";
       }
     ];
 

